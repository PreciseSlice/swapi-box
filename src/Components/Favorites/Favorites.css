.favoriteContainer {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  overflow: auto;
  flex-wrap: wrap;
  justify-content: space-evenly; }

.noFavorites {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  text-align: center; }
  .noFavorites h1 {
    color: #feda4a;
    font-size: 60px; }
