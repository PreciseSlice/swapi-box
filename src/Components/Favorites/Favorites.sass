.favoriteContainer
  height: 100%
  width: 100% 
  display: flex
  flex-direction: row
  overflow: auto
  flex-wrap: wrap
  justify-content: space-evenly

.noFavorites
  display: flex
  height: 100%
  justify-content: center
  margin-top: 30%
  text-align: center

  h1
    color: #feda4a
    font-size: 4em

@media (max-width: 500px)
  .noFavorites h1
    font-size: 2.8em
