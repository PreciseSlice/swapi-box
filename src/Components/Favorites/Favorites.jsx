import React from 'react';
import './Favorites.css';
import Card from '../Card/Card.jsx';
import PropTypes from 'prop-types';

const Favorites = ({ favorites }) => {
  const renderCards = favorites.map(favorite => (
    <Card data={favorite} key={favorite.name} />
  ));
  return <div className="favoriteContainer">{renderCards}</div>;
};

Favorites.propTypes = {
  favorites: PropTypes.array.isRequired
};

export default Favorites;
