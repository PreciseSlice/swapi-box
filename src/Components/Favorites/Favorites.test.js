/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Favorites from './Favorites';
import { shallow, mount } from 'enzyme';

describe('Favorites', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Favorites
        favorites={[
          {
            name: 'Alderaan',
            climate: 'temperate',
            terrain: 'grasslands, mountains',
            population: '2000000000'
          },
          {
            homeworld: 'planet',
            name: 'Luke Skywalker',
            population: 1000000,
            species: ['human']
          },
          {
            homeworld: 'planet',
            name: 'C-3PO',
            population: 100000,
            species: ['droid']
          }
        ]}
        clickHandler={jest.fn()}
      />
    );
  });

  describe('Favorites', () => {
    it('Exist and matches snapshot', () => {
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });

    it('renders a card for each object in the array it is passed', () => {
      expect(wrapper.find('.card').length).toEqual(3);
    });
  });
});
