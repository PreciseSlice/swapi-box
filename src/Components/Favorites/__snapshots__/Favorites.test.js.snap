// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Favorites Exist and matches snapshot 1`] = `
<Favorites
  favClickHandler={[Function]}
  favorites={
    Array [
      Object {
        "climate": "temperate",
        "name": "Alderaan",
        "population": "2000000000",
        "terrain": "grasslands, mountains",
      },
      Object {
        "homeworld": "planet",
        "name": "Luke Skywalker",
        "population": 1000000,
        "species": Array [
          "human",
        ],
      },
      Object {
        "homeworld": "planet",
        "name": "C-3PO",
        "population": 100000,
        "species": Array [
          "droid",
        ],
      },
    ]
  }
>
  <div
    className="favoriteContainer"
  >
    <Card
      favClickHandler={[Function]}
      data={
        Object {
          "climate": "temperate",
          "name": "Alderaan",
          "population": "2000000000",
          "terrain": "grasslands, mountains",
        }
      }
      key="Alderaan"
    >
      <div
        className="card undefined"
      >
        <h1>
          Alderaan
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: temperate
          </li>
          <li
            key="function now() { [native code] }1"
          >
            terrain: grasslands, mountains
          </li>
          <li
            key="function now() { [native code] }2"
          >
            population: 2000000000
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      favClickHandler={[Function]}
      data={
        Object {
          "homeworld": "planet",
          "name": "Luke Skywalker",
          "population": 1000000,
          "species": Array [
            "human",
          ],
        }
      }
      key="Luke Skywalker"
    >
      <div
        className="card undefined"
      >
        <h1>
          Luke Skywalker
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            homeworld: planet
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 1000000
          </li>
          <li
            key="function now() { [native code] }2"
          >
            species: human
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      favClickHandler={[Function]}
      data={
        Object {
          "homeworld": "planet",
          "name": "C-3PO",
          "population": 100000,
          "species": Array [
            "droid",
          ],
        }
      }
      key="C-3PO"
    >
      <div
        className="card undefined"
      >
        <h1>
          C-3PO
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            homeworld: planet
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 100000
          </li>
          <li
            key="function now() { [native code] }2"
          >
            species: droid
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
  </div>
</Favorites>
`;
